### DISCUSSION TITLE
CONTINUOUS vs. ON_DEMAND updating

### DISCUSSION THREAD

**Discussion by miles-tbe** [pelicanmapping/osgearth#1860] [Support Q&A] [answered]
Hi,
I really want to know just how these two run frame schemes work so I know which would be best for our project. Currently we use `Viewer.addRunFrameSceheme (Viewer.CONTINUOUS);` and `Viewer.addUpdateOperation (new UpdateOp ());` as well as an EventHandler for other items. 

1) If we changed to ON_DEMAND would the UpdateOperation need to change in some way to work? 
2) Are one of these UpdateOperation/EventHandler functions necessary in order to have things update at all or are there other way that you can trigger the viewer to update using just regular functions we create outselves? 

Thanks,
Miles

**Comment by gwaldron** [comment, marked as answer]
Miles,
osgEarth 3.x only supports the CONTINUOUS run scheme right now.

Usually when using ON_DEMAND you need to trigger a redraw through the osgGA::GUIActionAdapter interface, which is available from an event handler. However OpenSceneGraph will disable ON_DEMAND mode if you install an update operation or any update callbacks on your scene graph.

**Reply by miles-tbe** [reply]
Thanks for the explanation.
